### PR TITLE
Dockerfile.template designed by hhorak and me.

### PR DESCRIPTION
**Pull request by phracek**
Signed-off-by: Petr "Stone" Hracek <phracek@redhat.com>

Together with Honza Horak, we have designed and Dockerfile.template which could help beginners to start with Docker.
@hhorak: Thanks for help.

I would be welcome for your feedback. Does it make sense for you? Please keep it mind, if you are docker newcomer, you will want some help at the beginning.

### CODE REVIEW

File: Dockerfile.template [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+# This exposes port from container to host. Specify it during `docker run` as parameter -p 25:10025
+# EXPOSE 10025
+
+# Specify username which will be used during running container
+# USER 89
+
+# VOLUME mounts host dir into container dir. It is specified by parameter -v <host_directory>"<container_directory>
+# VOLUME ['/var/spool/postfix']
+
+# Command which will be start service during command `docker run`
```
Comment (TomasTomecek) [inline Dockerfile.template]: `which will start the service`

File: Dockerfile.template [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+# Command which runs during `docker build`
+# RUN /files/<some_bash_script>.sh
+
+# This exposes port from container to host. Specify it during `docker run` as parameter -p 25:10025
+# EXPOSE 10025
+
+# Specify username which will be used during running container
+# USER 89
+
+# VOLUME mounts host dir into container dir. It is specified by parameter -v <host_directory>"<container_directory>
```
Comment (TomasTomecek) [inline Dockerfile.template]: `<host_dir>:<cont_dir>`, not `"`

File: Dockerfile.template [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+# ADD scripts /files
+
+# Command which runs during `docker build`
+# RUN /files/<some_bash_script>.sh
+
+# This exposes port from container to host. Specify it during `docker run` as parameter -p 25:10025
+# EXPOSE 10025
+
+# Specify username which will be used during running container
+# USER 89
```
Comment (TomasTomecek) [inline Dockerfile.template]: I'm assuming that the user is created when postfix package is installed, hence I would prefer name over uid. Also we should cover scenario, when the "user account" is not created.

File: Dockerfile.template [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+# Additional packages
+#  * findutils are needed in case fedora:24 is loaded from docker.io.
+
+# RUN dnf install -y --setopt=tsflags=nodocs <your_package_names> && \
+    dnf -y clean all
+
+# MAINTAINER "real_name" <email_adress>
+
+# LABEL summary="Postfix is a Mail Transport Agent (MTA)." \
+#       description="Some further description." \
```
Comment (TomasTomecek) [inline Dockerfile.template]: Can you please fill the further description since this is a real example?

File: Dockerfile.template [hunk 5]
```diff
@@ -0,0 +1,46 @@
+FROM fedora:latest
+
+# Provide some Dockerfile description, like is mentioned below
```
Comment (TomasTomecek) [inline Dockerfile.template:3]: I like this section, would be awesome if we had this as a machine readable metadata.

**Review by TomasTomecek — CHANGES_REQUESTED** [review]
A couple nits, one typo, one grammar issue; nothing important. Nicely done!

**Comment by eliskasl** [comment]
I didn't get the commenting :) How about removing comments (#) from every line that is actually not a comment? It would make it a bit more readable. Since it's a template I guess nobody will actually attempt to run it :)
Anyway good work, I like it.

**Comment by eliskasl** [comment]
Btw, you might want to tag others when requesting feedback, not sure if anyone outside the Admin group gets a notification when a PR is created.

**Comment by TomasTomecek** [comment]
> Btw, you might want to tag others when requesting feedback, not sure if anyone outside the Admin group gets a notification when a PR is created.

8 people watch this repo, so hopefully they are getting the notifications. Definitively a good suggestion though.

File: Dockerfile.template (hunk 5 above)
Comment (phracek) [reply Dockerfile.template:3]: An example would be welcome, I will do it. Thanks for tips and review.

**Comment by phracek** [comment]
@whitel @james-antill @asamalik Hi folks, I did a couple of updates. Feel free to comment it.

File: Dockerfile.template [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+RUN /files/<some_bash_script>.sh
+
+# This exposes port from container to host. Specify it during `docker run` as parameter -p 25:10025
+EXPOSE 10025
+
+# Specify username which will be used during running container
+USER postfix
+
+# VOLUME mounts host dir into container dir. It is specified by parameter -v <host_directory>:<container_directory>
+VOLUME ['/var/spool/postfix']
```
Comment (TomasTomecek) [inline Dockerfile.template:46]: I know that I'm such a nitpicker -- I should have mention it in previous round, I'm sorry about that.

This is not entirely true. `VOLUME` instruction creates unnamed volume in `/var/lib/docker/volumes` and mounts it inside a container -- so your data do NOT persist in the container. Doing `docker run -v ...` is just an override. Hence, could we update the comment to:

```
VOLUME instruction creates unnamed volume and mounts it to the provided path, you can override this behavior by mounting a selected host directory into container: "-v <host_directory>:<container_directory>"
```

File: Dockerfile.template [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+# Variable which is posted to RUN and CMD directive
+ENV POSTFIX_SMTP_PORT=10025
+
+# Directive adds files from a directory (e.g. scripts) in current directory to a container directory like /files
+ADD scripts /files
+
+# Command which runs during `docker build`
+RUN /files/<some_bash_script>.sh
+
+# This exposes port from container to host. Specify it during `docker run` as parameter -p 25:10025
```
Comment (TomasTomecek) [inline Dockerfile.template]: Please also mention that the format is `<host>:<container>`.

**Review by TomasTomecek — COMMENTED** [review]
2 more tiny docs nits. Otherwise LGTM, nicely done, Peťo!

File: Dockerfile.template (hunk 6 above)
Comment (phracek) [reply Dockerfile.template:46]: No problem. I would like to create a template which can be useful for the others. So better explanation is welcome.

**Review by TomasTomecek — APPROVED** [review]

**Pull request merged**